tools/lang/examples/l3.dscp

```diff
@@ -0,0 +1,192 @@
+/*
+Token specification example from l3. See https://github.com/digicatapult/dscp-documentation/blob/main/docs/l3/index.md for details
+*/
+
+// `token`s define token formats that should be persisted in transitions
+token Demand {
+  // fields can be of type Role, File, Literal, a specific literal, a token type or a union of any of these
+  owner: Role,
+  subtype: "demand_a" | "demand_b",
+  state: "created" | "allocated" | "cancelled",
+}
+
+token Match2 {
+  optimiser: Role,
+  member_a: Role,
+  demand_a: Demand,
+  member_b: Role,
+  demand_b: Demand,
+  state: "proposed" | "acceptedA" | "acceptedB" | "acceptedFinal" | "cancelled",
+  replaces: Match2 | None,
+}
+
+// `fn`s define process flow restrictions and consist of a set of inputs, a set of outputs and a set of additional conditions
+// A `fn` can be private in which case it will not be published as a flow itself but can be referred to in the conditions of
+// other flows
+fn clone_demand_partial | a: Demand | => | b: Demand | where {
+  a == b,
+  a.owner == b.owner,
+  a.subtype == b.subtype
+}
+
+fn clone_demand | a: Demand | => | b: Demand | where {
+  clone_demand_partial | a | => | b |,
+  a.state == b.state,
+}
+
+fn clone_match_partial | a: Match2 | => | b: Match2 | where {
+  a == b,
+  a.optimiser == b.optimiser,
+  a.member_a == b.member_a,
+  a.demand_a == b.demand_a,
+  a.member_b == b.member_b,
+  a.demand_b == b.demand_b,
+  a.replaces == b.replaces,
+}
+
+fn clone_match | foo: Match2 | => | bar: Match2 | where {
```

Comment:
did you want to switch this back to 'a' and 'b' instead of 'foo' and 'bar'? Is just a cosmetic issue

Reply:
yep good spot